File: apps/api/app/schemas/analytics.py
"""Analytics schemas."""
from typing import List
from pydantic import BaseModel


class VelocityPoint(BaseModel):
    sprint_name: str
    completed_points: int
    planned_points: int


class BurndownPoint(BaseModel):
    day: str
    remaining: int
    ideal: float


class TaskStatusDistribution(BaseModel):
    status: str
    count: int


class TeamMemberUtilization(BaseModel):
    user_id: str
    full_name: str
    open_tasks: int
    hours_logged: float


class ProjectAnalytics(BaseModel):
    total_tasks: int
    completed_tasks: int
    open_tasks: int
    overdue_tasks: int
    velocity: List[VelocityPoint]
    status_distribution: List[TaskStatusDistribution]
    team_utilization: List[TeamMemberUtilization]

File: apps/apps/web/src/api/analytics.ts
import { apiClient } from '../lib/auth'

export interface ProjectAnalytics {
  total_tasks: number
  completed_tasks: number
  open_tasks: number
  overdue_tasks: number
  velocity: Array<{ sprint_name: string; completed_points: number; planned_points: number }>
  status_distribution: Array<{ status: string; count: number }>
  team_utilization: Array<{ user_id: string; full_name: string; open_tasks: number; hours_logged: number }>
}

export const analyticsApi = {
  projectStats: (projectId: string, orgId: string) =>
    apiClient.get<{ data: ProjectAnalytics }>(`/analytics/projects/${projectId}?org_id=${orgId}`).then(r => r.data.data),
}
